src/AWS.Deploy.Common/Recipes/OptionSettingValidation.cs

```diff
@@ -11,8 +11,21 @@ public enum ValidationStatus
 
     public class OptionSettingValidation
     {
+        /// <summary>
+        /// Determines whether the current value as set by the user is in a valid or invalid state.
+        /// </summary>
         public ValidationStatus ValidationStatus { get; set; } = ValidationStatus.Valid;
 
+        /// <summary>
+        /// The validation message in the case where the value set by the user is an invalid one.
+        /// This is empty in the case where the value set by the user is a valid one.
+        /// </summary>
         public string ValidationMessage { get; set; } = string.Empty;
+
+        /// <summary>
+        /// The value last attempted to be set by the user in the case where that value is invalid.
+        /// This is null in the case where the value is valid.
+        /// </summary>
+        public object? InvalidValue { get; set; }
```

Comment:
Curious on @awschristou's take on this if he gets a chance - is it awkward that any wrapper tools will have to do something like this?
```
if (optionSettingItemSummary.Validation.ValidationStatus == Valid)
   Display optionSettingItemSummary.Value
else
  Display optionSettingItemSummary.Validation.InvalidValue

Should the server be more opinionated about _"if you are showing a form to the user, here is the value to show"_?

Reply:
I agree it is good for clients to have access to both values (last-valid and current-invalid). In the current Toolkit implementation, the "last good" value will be ignored, but this PR's design allows clients to stay in sync with the backend. This is why I don't mind the conditional check mentioned by @ashovlin .

Both the clients and the backend shouldn't allow a deployment if there are invalid values as well.